frontend/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,56 @@
+import { BiCaretDown, BiCaretUp } from 'react-icons/bi'
+import {
+  forwardRef,
+  Icon,
+  InputRightElement,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputProps as ChakraNumberInputProps,
+  NumberInputStepper,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { omit } from '@chakra-ui/utils'
+
+import { BxsCheckCircle } from '~assets/icons/BxsCheckCircle'
+
+export interface NumberInputProps extends ChakraNumberInputProps {
+  /**
+   * Whether the input is in a prefilled state.
+   */
+  isPrefilled?: boolean
+  /**
+   * Whether the input is in a success state.
+   */
+  isSuccess?: boolean
+}
+
+export const NumberInput = forwardRef<NumberInputProps, 'input'>(
+  (props, ref) => {
+    const { isSuccess } = props
+    const inputStyles = useMultiStyleConfig('NumberInput', props)
+    // Omit extra props so they will not be passed into the DOM and trigger
+    // React warnings.
+    const inputProps = omit(props, ['isSuccess', 'isPrefilled'])
```

Comment:
would it be better to remove the prop totally as the `isPrefilled` prop is omitted here and not used anywhere

Reply:
inside the theme :P